packages/renderer/src/lib/image/PullImage.spec.ts

```diff
@@ -64,48 +64,40 @@ beforeAll(() => {
   });
 });
 
+const CONTAINER_CONNECTION_MOCK: ProviderContainerConnectionInfo = {
+  name: 'test',
+  displayName: 'test',
+  status: 'started',
+  endpoint: {
+    socketPath: '',
+  },
+  type: 'podman',
+};
+
+const PROVIDER_INFO_MOCK: ProviderInfo = {
+  id: 'test',
+  internalId: 'id',
+  name: '',
+  containerConnections: [CONTAINER_CONNECTION_MOCK],
+  kubernetesConnections: [],
+  status: 'started',
+  containerProviderConnectionCreation: false,
+  containerProviderConnectionInitialization: false,
+  kubernetesProviderConnectionCreation: false,
+  kubernetesProviderConnectionInitialization: false,
+} as unknown as ProviderInfo;
+
 beforeEach(() => {
   vi.resetAllMocks();
   vi.restoreAllMocks();
+
+  providerInfos.set([PROVIDER_INFO_MOCK]);
 });
 
 const buttonText = 'Pull image';
 
-// the pull image page expects to have a valid provider connection, so let's mock one
-function setup() {
```

Comment:
Every test had the `setup` call, let's moving the logic to the `beforeEach` (it simply the tests I added to move it to the beforeEach)